src/server/json_family.cc

```diff
@@ -259,8 +279,108 @@ OpResult<vector<OptBool>> OpToggle(const OpArgs& op_args, string_view key, strin
   return vec;
 }
 
+template <typename Op>
+OpResult<string> OpDoubleArithmetic(const OpArgs& op_args, string_view key, string_view path,
+                                    double num, Op arithmetic_op) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  bool is_result_overflow = false;
+  double int_part;
+  bool has_fractional_part = (modf(num, &int_part) != 0);
+  json output(json_array_arg);
+
+  auto cb = [&](const string& path, json& val) {
+    if (val.is_number()) {
+      double result = arithmetic_op(val.as<double>(), num);
+      if (isinf(result)) {
+        is_result_overflow = true;
+      }
+
+      if (val.is_double() || has_fractional_part) {
+        val = result;
+      } else {
+        val = (uint64_t)result;
+      }
+      output.push_back(val);
+    } else {
+      output.push_back(json::null());
+    }
+  };
+
+  json j = result.value();
+  error_code ec = JsonReplace(j, path, cb);
```

Comment:
i think there is a bug here when there are multiple matches and just one of them overflows. Then you keep the system in an inconsistent state (i.e change something but return an overflow error). I do not see how it can be solved without running the query first checking everything works, and then run replace. Also, please add a test covering this case.

Reply:
I thought also about this case, but there isn't a bug here. The code commits the JSON changes at the line starting with `SetString` this occurs after the check if the numeric result overflows.